### PR TITLE
Configure in parallel

### PR DESCRIPTION
**Pull request by mmichal10**

### CODE REVIEW

**Comment by KamilLepek** [comment]
retest this please

File: configure [hunk 1]
```diff
@@ -3,23 +3,71 @@
...
+CONFIG_FILE="config.out"
+
+generate_config() {
+	rm -f ${CONFIG_FILE}
+	touch ${CONFIG_FILE}
+	declare -a pid_list
+	progress=0
+
+	# Compile each test module in backgorund
+	echo "Prepraing configuration"
```
Comment (KamilLepek) [inline configure]: Preparing

File: configure [hunk 2]
```diff
@@ -3,23 +3,71 @@
...
+
+	# Wait for all compilation processess to finish
+	for i in "${pid_list[@]}"; do
+		wait $i &> /dev/null
+	done
 
-rm -f $SCRIPTPATH/modules/generated_defines.h
+	grep "X" ${SCRIPTPATH}/${CONFIG_FILE} &> /dev/null
+	if [ $? -eq 0 ] ; then
+		echo "ERROR! Follwoing steps failed while preparing config:"
```
Comment (KamilLepek) [inline configure]: Following

File: configure [hunk 3]
```diff
@@ -3,23 +3,71 @@
...
+	progress=0
+
+	# Compile each test module in backgorund
+	echo "Prepraing configuration"
+		for file in $CONFIG_FILES; do
+			progress=$((progress+1))
+			# $1 - Action to be performed
+			# $2 - File with stored configuration
+			# $3 - Name of called script (since script is runed as subprocess
+			#		it has to be passed explicite
```
Comment (KamilLepek) [inline configure]: explicitly)

File: configure [hunk 4]
```diff
@@ -3,23 +3,71 @@
...
+	declare -a pid_list
+	progress=0
+
+	# Compile each test module in backgorund
+	echo "Prepraing configuration"
+		for file in $CONFIG_FILES; do
+			progress=$((progress+1))
+			# $1 - Action to be performed
+			# $2 - File with stored configuration
+			# $3 - Name of called script (since script is runed as subprocess
```
Comment (KamilLepek) [inline configure]: maybe "is running"?

File: configure [hunk 5]
```diff
@@ -3,23 +3,71 @@
...
+			#		it has to be passed explicite
+			source $file "check" "${SCRIPTPATH}/${CONFIG_FILE}" "$file" &
+			pid_list+=($!)
+	done
+
+	# Wait for all compilation processess to finish
```
Comment (KamilLepek) [inline configure]: processes

**Comment by mmichal10** [comment]
retest this please

**Pull request closed without merging**